apps/services/user-profile/src/app/user-profile/verification.service.ts

```diff
@@ -107,8 +108,11 @@ export class VerificationService {
   async confirmEmail(
     confirmEmailDto: ConfirmEmailDto,
     nationalId: string,
-    transaction?: Transaction,
+    options: { transaction?: Transaction; maxRetries?: number } = {
+      maxRetries: EMAIL_VERIFICATION_MAX_TRIES,
+    },
   ): Promise<ConfirmationDtoResponse> {
+    const { transaction, maxRetries = EMAIL_VERIFICATION_MAX_TRIES } = options
```

Comment:
I tried that first but had issues with that because the whole object can be nuill that why I need the default object on the argument input level, and it can also only have transaction and not the maxRetries, thats why I need to object destruction. 
If I skip the object destruction in line 115 and send in `{transaction: transaction}` it will not initialize `maxRetries` with the initial value.

Reply:
I haven't tried skipping the default values on the argument level though and only use the object destruction. That would probably work, and is much cleaner